lib/connect.c

```diff
@@ -409,7 +409,7 @@ static CURLcode bindlocal(struct Curl_easy *data,
                Curl_printable_address. The latter returns only numeric scope
                IDs and the former returns none at all.  So the scope ID, if
                present, is known to be numeric */
-            si6->sin6_scope_id = atoi(scope_ptr);
+            si6->sin6_scope_id = (unsigned int)strtoul(scope_ptr, NULL, 10);
```

Comment:
if we change this, shouldn't we also make a proper overflow check as well? After all, `strtoul()` might return a value that cannot be stored as a scope id...